cmd/sigstore-go/main.go

```diff
@@ -214,3 +234,214 @@ func trustedPublicKeyMaterial(pk crypto.PublicKey) *root.TrustedPublicKeyMateria
+		return nil, fmt.Errorf("error getting signing certificate: %w", err)
+	}
+	// 2. Get the transparency log entries
+	tlogEntries, err := getVerificationMaterialTlogEntries(manifestLayer)
+	if err != nil {
+		return nil, fmt.Errorf("error getting tlog entries: %w", err)
+	}
+	// 3. Construct the verification material
+	return &protobundle.VerificationMaterial{
+		Content:                   signingCert,
+		TlogEntries:               tlogEntries,
+		TimestampVerificationData: nil,
+	}, nil
+}
+
+// getVerificationMaterialTlogEntries returns the verification material transparency log entries from the simple signing layer
+func getVerificationMaterialTlogEntries(manifestLayer *v1.Descriptor) ([]*protorekor.TransparencyLogEntry, error) {
+	// 1. Get the bundle annotation
+	bun := manifestLayer.Annotations["dev.sigstore.cosign/bundle"]
+	var jsonData map[string]interface{}
+	err := json.Unmarshal([]byte(bun), &jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling json: %w", err)
+	}
+	// 2. Get the log index, log ID, integrated time, signed entry timestamp and body
+	logIndex, ok := jsonData["Payload"].(map[string]interface{})["logIndex"].(float64)
+	if !ok {
+		return nil, fmt.Errorf("error getting logIndex")
+	}
+	li, ok := jsonData["Payload"].(map[string]interface{})["logID"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting logID")
+	}
+	logID, err := hex.DecodeString(li)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding logID: %w", err)
+	}
+	integratedTime, ok := jsonData["Payload"].(map[string]interface{})["integratedTime"].(float64)
+	if !ok {
+		return nil, fmt.Errorf("error getting integratedTime")
+	}
+	set, ok := jsonData["SignedEntryTimestamp"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting SignedEntryTimestamp")
+	}
+	signedEntryTimestamp, err := base64.StdEncoding.DecodeString(set)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding signedEntryTimestamp: %w", err)
+	}
+	// 3. Unmarshal the body and extract the rekor KindVersion details
+	body, ok := jsonData["Payload"].(map[string]interface{})["body"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting body")
+	}
+	bodyBytes, err := base64.StdEncoding.DecodeString(body)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding body: %w", err)
+	}
+	err = json.Unmarshal(bodyBytes, &jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling json: %w", err)
+	}
+	apiVersion := jsonData["apiVersion"].(string)
+	kind := jsonData["kind"].(string)
+	// 4. Construct the transparency log entry list
+	return []*protorekor.TransparencyLogEntry{
+		{
+			LogIndex: int64(logIndex),
+			LogId: &protocommon.LogId{
+				KeyId: logID,
+			},
+			KindVersion: &protorekor.KindVersion{
+				Kind:    kind,
+				Version: apiVersion,
+			},
+			IntegratedTime: int64(integratedTime),
+			InclusionPromise: &protorekor.InclusionPromise{
+				SignedEntryTimestamp: signedEntryTimestamp,
+			},
+			InclusionProof:    nil,
```

Comment:
You're right that we don't store the inclusion proof - https://github.com/sigstore/cosign/blob/main/pkg/cosign/bundle/rekor.go#L23-L34

Ideally we would be storing the inclusion proof too. For some history, it's because we previously weren't doing anything with inclusion proofs, it's only been recently that we're improving the offline verification from rekor.

I'd rather not make the change in Cosign since the "rekor bundle" format has been stable for some time now. Instead we should implement protobuf bundle support which will include the inclusion proof, since it'll be stored after signing.

I'd personally prefer to not have more v01 bundles without these stronger proofs and instead make the change in Cosign rather than try to translate the existing OCI contents into an old bundle format.